react-charts/sparkline-chart/step3.md

```diff
@@ -1,56 +1,52 @@
-
-2) Next, we'll set up the general structure of the chart and the necessary import statements.
-
-Copy the following code into the App.js file, replacing all of the content there:
-
-<pre class="file" data-filename="App.js" data-target="replace">
-import React from 'react';
-import "@patternfly/react-core/dist/styles/base.css";
-import './app.css';
-import {
-  ChartArea,
-  ChartContainer,
-  ChartGroup,
-  ChartLabel,
-  ChartThemeColor,
-  ChartVoronoiContainer
-} from '@patternfly/react-charts';
-
-function App() {
-  return (
-    &lt;div className=&quot;chart-label-container&quot;&gt;
-      &lt;div className=&quot;chart-container&quot;&gt;
-        &lt;ChartGroup
-          maxDomain={{y: 9}}
-          padding={0}
-          height={100}
-          width={400}
-        &gt;
-          &lt;ChartArea
-            data={[
-              { name: &#39;Cats&#39;, x: &#39;2015&#39;, y: 3 }, 
-              { name: &#39;Cats&#39;, x: &#39;2016&#39;, y: 4 }, 
-              { name: &#39;Cats&#39;, x: &#39;2017&#39;, y: 8 }, 
-              { name: &#39;Cats&#39;, x: &#39;2018&#39;, y: 6 }
-            ]}
-          /&gt;
-        &lt;/ChartGroup&gt;
-      &lt;/div&gt;
-    &lt;/div&gt;
-  );
-}
-
-export default App;
-</pre>
-
-- The `maxDomain` property defines the maximum domain value for a chart
-
-3) When the server reloads, you should see something like this:
-<img src="sparkline-chart/assets/simple.png" alt="Simple chart" style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
-
-We'll continue by adding tooltips to the chart in the next step.
+Add tooltips to the sparkline chart.
+
+1) <strong>Make sure `App.js` is still open.</strong>
+
+2) <strong>Locate the code within the `<ChartGroup/>` component.</strong>
+
+It should look like this:
+
+<pre class="file">
+&lt;ChartGroup
+  maxDomain={{y: 9}}
+  padding={0}
+  height={100}
+  width={400}
+&gt;
+</pre>
+
+3) <strong>Add a `containerComponent` property to the `ChartGroup` component.</strong>
+
+The `containerComponent` property takes a component instance, which will be used
+to create a container element for standalone charts.
+Inside the `containerComponent` is the `ChartVoronoiContainer` component, which
+adds the ability to associate a mouse position with the data point(s) closest to
+it. This is useful for adding a tooltip.
+
+<pre class="file" data-target="clipboard">
+containerComponent={
+  &lt;ChartVoronoiContainer
+    constrainToVisibleArea
+    labels={({ datum }) =&gt; `${datum.name}: ${datum.y}`}
+  /&gt;
+}
+</pre>
+
+4) <strong>Add a `constrainToVisibleArea` property inside of the `ChartVoronoiContainer`.</strong>
+
+The `constrainToVisibleArea` property determines whether to coerce tooltips so
+that they fit within the visible area of the chart.
+
+It should look like this:
+
+<pre class="file" data-target="clipboard">
+containerComponent={
+  &lt;ChartVoronoiContainer
+    constrainToVisibleArea
+  /&gt;
+}
+</pre>
+

```

Comment:
Add step 5 for the labels, to that its

containerComponent={
  <ChartVoronoiContainer
    constrainToVisibleArea
    labels={({ datum }) => `${datum.name}: ${datum.y}`}
  />
}

look at area-chart: step 4, no.5